src/include/codegen/stage.h

```diff
@@ -0,0 +1,35 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// stage.h
+//
+// Identification: src/include/codegen/stage.h
+//
+// Copyright (c) 2015-2017, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "codegen/codegen.h"
+
+namespace peloton {
+namespace codegen {
+
+enum class StageKind {
+  SINGLE_THREADED,
+};
+
+struct CodeGenStage {
+  StageKind kind_;
+  llvm::Function *llvm_func_;
+};
+
+struct Stage {
```

Comment:
What are we expecting these for?

Reply:
In the next PR, we will have a different kind of stage - multi-threaded scan. Going forward, we could also have multi-thread aggregate stage.

For the multi-threaded scan stage, we have a single task function, and rely on `ExecuteStages` to submit tasks to the queue. Note that this means:
- Each "stage function" invocation happens in a single thread, so we don't need to do synchronization stuff within LLVM;
- Submitting tasks is all done in C++, which makes code more clear (hopefully?) and exception handling easier (because we can just wrap the call to task functions with try-catch).